packages/node_modules/@node-red/editor-client/src/js/ui/editor.js

```diff
@@ -540,7 +540,113 @@ RED.editor = (function() {
         return label;
     }
 
-    function buildEditForm(container,formId,type,ns) {
+    function buildEnvForm(container, node) {
+        var env_container = $('#node-input-env-container');
+        env_container
+            .css('min-height','250px').css('min-width','450px')
+            .editableList({
+                addItem: function(container, i, opt) {
+                    var row = $('<div/>').appendTo(container);
+                    var nameField = $('<input/>', {
+                        class: "node-input-env-name",
+                        type: "text",
+                        style: "margin-length: 5px; width: 32%;"
+                    }).appendTo(row);
+                    var valueField = $('<input/>',{
+                        class: "node-input-env-value",
+                        type: "text",
+                        style: "margin-left: 5px; width: 65%;"
+                    }).appendTo(row)
+                    valueField.typedInput({default:'str',
+                                           types:['str','num','bool','json','bin','re','date']
+                                          });
+                    if (opt) {
+                        nameField.val(opt.name);
+                        valueField.typedInput('type', opt.type);
+                        valueField.typedInput('value', opt.value);
+                    }
+                },
+                sortable: true,
+                removable: true
+            });
+        var envs = node.env;
+        if (envs) {
+            for (var i = 0; i < envs.length; i++) {
+                var env = envs[i];
+                env_container.editableList('addItem', env);
+            }
+        }
+    }
+
+    function convEnv(editable_list) {
+        if (editable_list) {
+            var env = [];
+            editable_list.each(function(i) {
+                var entry = $(this);
+                var nf = entry.find(".node-input-env-name");
+                var vf = entry.find(".node-input-env-value");
+                var name = nf.val();
+                var value = vf.typedInput("value");
+                var type = vf.typedInput("type");
+                var info = {
+                    name: name,
+                    label: "",
+                    value: value,
+                    type: type,
+                    target_type: "env var",
+                    target: name
+                };
+                var item = {
+                    name: name,
+                    type: type,
+                    value: value,
+                    info: info
+                };
+                env.push(item);
+            });
+            return env;
+        }
+        return null;
+    }
+
+    function isSameEnv(env0, env1) {
```

Comment:
Can we use `JSON.stringify` of `env0` and `env1` to do a simple String comparison?